### PR TITLE
tarsnap: add package option

### PR DESCRIPTION
**Pull request by wamserma**
###### Description of changes

Add an option to select the used package for the tarsnap service. This is useful to choose tarsnap from an package set where unfree packages have been enabled.
Verified to work in production. ;)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/backup/tarsnap.nix [hunk 1]
```diff
@@ -32,6 +32,13 @@ in
     services.tarsnap = {
       enable = mkEnableOption (lib.mdDoc "periodic tarsnap backups");
 
+      package = mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/backup/tarsnap.nix:35]: Please use mkPackageOption

File: nixos/modules/services/backup/tarsnap.nix [hunk 2]
```diff
@@ -318,7 +325,7 @@ in
         '';
 
         script = let
-          tarsnap = ''tarsnap --configfile "/etc/tarsnap/${name}.conf"'';
+          tarsnap = ''${gcfg.package}/bin/tarsnap --configfile "/etc/tarsnap/${name}.conf"'';
```
Comment (SuperSandro2000) [inline nixos/modules/services/backup/tarsnap.nix:328]:
```suggestion
          tarsnap = ''${lib.getExe gcfg.package} --configfile "/etc/tarsnap/${name}.conf"'';
```
we could utilize meta.mainProgram here

Comment (wamserma) [reply nixos/modules/services/backup/tarsnap.nix]: also added meta.mainProgram to the tarsnap package

File: pkgs/tools/backup/tarsnap/default.nix [hunk 3]
```diff
@@ -39,6 +39,7 @@ stdenv.mkDerivation rec {
     homepage    = "http://www.tarsnap.com/";
     license     = lib.licenses.unfree;
     platforms   = lib.platforms.unix;
+    mainProgram = pname;
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/tarsnap/default.nix:42]:
```suggestion
```
that's the default, no need to set it

**Pull request merged**